src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2245,7 +2245,7 @@ internal static void UpdateTypeChecking(this FileContentManager file, Compilatio
             }
         }
 
-        private static void CheckForGlobalCycleChange(FileContentManager file, List<Diagnostic> diagnostics)
+        private static void CheckForGlobalCycleChange(FileContentManager file, List<Diagnostic>? diagnostics)
```

Comment:
It looks like maybe `Count` and `Any` haven't been updated to use nullable reference types, but it says they throw `ArgumentNullException` if the source enumerable is null, so I think `diagnostics.Count(...)` and `diagnostics.Any(...)` where `diagnostics` is nullable is not safe.

Reply:
I've updated the method to have `diagnostics` be non-null here.